src/Orleans.Providers.Legacy.Runtime/Hosting/LegecyProviderSiloBuilderExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Orleans.Runtime;
+using Orleans.Configuration;
+using Orleans.Providers;
+using Orleans.Providers.Legacy;
+
+namespace Orleans.Hosting
+{
+    public static class LegecyProviderSiloBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the silo to use legacy Bootstrap providers
+        /// </summary>
+        public static ISiloHostBuilder UseBootstrapProviders(this ISiloHostBuilder builder)
+        {
+            return builder.ConfigureServices(services => services.UseBootstrapProviders());
+        }
+
+        /// <summary>
+        /// Configures the silo to use legacy Bootstrap providers
+        /// </summary>
+        public static IServiceCollection UseBootstrapProviders(this IServiceCollection services)
+        {
+            services.AddSingleton<BootstrapProviderLifecycle>();
+            services.AddFromExisting<ILifecycleParticipant<ISiloLifecycle>, BootstrapProviderLifecycle>();
+            services.AddSingleton<BootstrapProviderManager>();
+            services.AddFromExisting<IProviderManager, BootstrapProviderManager>();
+            return services;
```

Comment:
we should attempt to make this method idempotent I believe

Reply:
Suggestions?
Use TryXXX calls?  Some other means?

Reply:
Will address, though I feel compelled to note we don't seem to be adding such protections for many core and more critical systems, so it's unclear why this is deemed relevant here.

Reply:
It is not strictly done everywhere, although we are still in the process of adding more and more of these extension methods that we expect *end users* to call, where they might call it multiple times, or even infrastructure (such as the Orleans Dashboard, not sure if it's a great example) might want to make sure bootstrap providers are enabled, and call it just in case.

In a lot of our registrations we use the `TryAddXXX` methods, which makes them idempotent, and I do flag code that doesn't do it if I happen to see it. Nevertheless, even if we miss a few, the conforming container behavior says that resolving a service makes the last one registered be resolved, basically making it less important which actual registration worked if they were similar.

But, and why here is very important to do it correctly, when resolving all instances of a service, then it's no longer "the last registration wins and gets resolved". All of them are. And `ILifecycleParticipant<ISiloLifecycle>` is an important example of why adding it twice would break things, as all instances will be resolved and started (even if naively we thought of them as singletons when writing the code).